### PR TITLE
DOCSP-17236 drawer pages

### PR DESCRIPTION
**Pull request by kyuan-mongodb**
## Pull Request Info
Added drawer pages for the following API pages: Bson, Core, Sync

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-17236

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-17236-SeparateAPIDocLinks/

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging links in the PR description updated?

### If your page documents a concept, does it meet the following criteria?

- [ ] Target the [Jasmin persona](https://drive.google.com/file/d/14FbBOLCVxwSP6M9BK4Nz1Ir9tzxT8_02/view)
- [ ] Target the [Lucas persona](https://drive.google.com/file/d/1J2vqJxo7ldv7OP_obA9Q-avf0o_ju4Lk/view)

### CODE REVIEW

File: source/api-documentation.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+.. default-domain:: mongodb
+
+.. toctree::
+   :titlesonly:
+   :maxdepth: 1
+
+    Bson API <http://mongodb.github.io/mongo-java-driver/4.3/apidocs/bson/index.html>
+    Core Driver API <http://mongodb.github.io/mongo-java-driver/4.3/apidocs/mongodb-driver-core/index.html>
+    Sync Driver API <http://mongodb.github.io/mongo-java-driver/4.3/apidocs/mongodb-driver-sync/index.html>
```
Comment (ccho-mongodb) [inline source/api-documentation.txt:13]: Would it be better to use the titles from the current Java docs?
https://mongodb.github.io/mongo-java-driver/4.2/apidocs/
E.g.
Bson
Core
Java Driver (modern API)
Java Driver (legacy API)
I think we also refer to the last one somewhere in the docs, so we might include that if so.

**Review by ccho-mongodb — CHANGES_REQUESTED** [review]

**Review by ccho-mongodb — APPROVED** [review]
LGTM!

**Comment by kyuan-mongodb** [comment]
Hi @jyemin and @rozza , We want to make sure you're okay with the naming of the drawer pages.

**Comment by jyemin** [comment]
If they match what we have currently, then LGTM

**Pull request merged**